### PR TITLE
RD-6584 Helm Charts for Microservices

### PR DESCRIPTION
**Pull request by adrian-polanczyk-codilime**

### CODE REVIEW

File: cloudify-services/templates/certs-creator.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
```
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify-services/templates/certs-creator.yaml:1]: RD-6808

File: cloudify-services/templates/composer-backend.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+apiVersion: v1
+kind: Pod
```
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify-services/templates/composer-backend.yaml:2]: RD-6809

File: cloudify-services/templates/composer-backend.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+metadata:
+  labels:
+    app: composer-backend
+  name: composer-backend
+spec:
+  containers:
+    - env:
+        - name: RESTSERVICE_ADDRESS
+          value: nginx
+      image: docker.io/library/composer_backend
```
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify-services/templates/composer-backend.yaml:12]: RD-6810

File: cloudify-services/templates/composer-backend.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: composer-backend
+spec:
+  ports:
+    - port: 3000
+  selector:
+    app: composer-backend
+  type: NodePort
```
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify-services/templates/composer-backend.yaml:39]: RD-6811

File: cloudify-services/templates/config.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+apiVersion: v1
```
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify-services/templates/config.yaml:1]: RD-6812

File: cloudify-services/templates/mgmtworker.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      imagePullPolicy: IfNotPresent
+      name: mgmtworker
+      resources: {}
+      volumeMounts:
+        - mountPath: /etc/cloudify/ssl
+          name: ssl
+          readOnly: true
+        - mountPath: /opt/mgmtworker/work
+          name: mgmtworker-workdir
+        - mountPath: /opt/manager/resources/cloudify/types
```
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify-services/templates/mgmtworker.yaml:17]: RD-6813

File: cloudify-services/values.yaml [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+service:
+  type: ClusterIP
+  port: 80
+
+ingress:
```
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify-services/values.yaml:43]: eg. 
```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
  name: ingress-auth
spec:
  rules:
  - http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: nginx
            port:
              number: 80
```

File: cloudify-services/templates/rest-service.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+spec:
+  containers:
+    - image: docker.io/library/cloudify-manager-rest_service
+      imagePullPolicy: IfNotPresent
+      name: rest-service
+      resources: {}
+      ports:
+        - containerPort: 8100
+      volumeMounts:
+        - mountPath: /opt/manager
```
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify-services/templates/rest-service.yaml:16]: turns out that yes /opt/manager doesn't need to be mounted here

File: cloudify-services/templates/pvc.yaml [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+kind: PersistentVolumeClaim
+metadata:
+  name: manager-config
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 100Mi
+
```
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify-services/templates/pvc.yaml:50]: it seems that it is indeed unnecessary

File: cloudify-services/values.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (tehasdf) [reply cloudify-services/values.yaml:43]: in practice, let's just copy the one from `cloudify-manager-worker/`

**Review by tehasdf — APPROVED** [review]

**Review by mateumann — APPROVED** [review]

**Review by tehasdf — DISMISSED** [review]

**Review by tehasdf — APPROVED** [review]

**Pull request merged**